components/script/dom/htmlformelement.rs

```diff
@@ -332,30 +343,36 @@ impl HTMLFormElement {
                 return;
             }
         }
-        // Step 6
-        let mut form_data = self.get_form_dataset(Some(submitter));
 
-        // Step 7
+        // Step 8
         let encoding = self.pick_encoding();
 
-        // Step 8
+        // Step 9
+        let mut form_data = match self.get_form_dataset(Some(submitter)) {
+            Some(form_data) => form_data,
+            None => return,
+        };
+
+        // TODO: Step 10. If form cannot navigate, then return.
```

Comment:
Could you implement this check? The algorithm at https://html.spec.whatwg.org/multipage/links.html#cannot-navigate is straightforward, and there are three uses of it in the form submissions algorithm.

Reply:
Sure! Let me try to implement it!

Reply:
Implemented in 038c83b